certbot/tests/hook_test.py

```diff
@@ -30,19 +30,37 @@ def test_it(self, mock_validate_hook):
         self.assertEqual("renew", types[-1])
 
 
-class ValidateHookTest(util.TempDirTestCase):
+class ValidateHookTest(test_util.TempDirTestCase):
     """Tests for certbot.hooks.validate_hook."""
 
     @classmethod
     def _call(cls, *args, **kwargs):
         from certbot.hooks import validate_hook
         return validate_hook(*args, **kwargs)
 
-    @util.broken_on_windows
     def test_not_executable(self):
         file_path = os.path.join(self.tempdir, "foo")
+
+        # On Windows a file created within Certbot will always have all permissions to the
+        # Administrators group set. Since the unit tests are typically executed under elevated
+        # privileges, it means that current user will always have effective execute rights on the
+        # hook script, and so the test will fail. To prevent that and represent a file created
+        # outside Certbot as typically a hook file is, we mock the _generate_dacl function in
+        # certbot.compat.filesystem to give rights only to the current user. This implies removing
+        # all ACEs except the first one from the DACL created by original _generate_dacl function.
+
+        from certbot.compat.filesystem import _generate_dacl
```

Comment:
Since this test relies on mocking the internals of `filesystem.is_executable`, I propose moving the majority of the code in this test to a new class in `certbot/tests/compat/filesystem_test.py` and updating the code here to just mock out `filesystem.is_executable` entirely with a comment about it being hard to test without a mock on Windows.

What do you think?

Reply:
I agree. Done.